pkg/cmd/adm/restart_test.go

```diff
@@ -1,216 +1,169 @@
+		},
+	}
+	for k, tc := range tests {
+		t.Run(k, func(t *testing.T) {
+			//given
+			namespacedName := types.NamespacedName{
+				Namespace: tc.namespace,
+				Name:      tc.name,
 			}
-			ctx := clicontext.NewCommandContext(term, newClient)
-
-			// when
-			err := restart(ctx, clusterName, "cool-deployment")
-
-			// then
-			require.Error(t, err)
-			fakeClient.MockGet = nil
-			AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 3)
-			assert.Equal(t, 0, numberOfUpdateCalls)
-		})
-
-		t.Run("restart fails - deployment not found for "+clusterName, func(t *testing.T) {
-			// given
-			deployment := newDeployment(namespacedName, 3)
-			newClient, fakeClient := NewFakeClients(t, deployment)
-			numberOfUpdateCalls := 0
-			fakeClient.MockUpdate = requireDeploymentBeingUpdated(t, fakeClient, namespacedName, 3, &numberOfUpdateCalls)
+			var rolloutGroupVersionEncoder = schema.GroupVersion{Group: "apps", Version: "v1"}
+			deployment1 := newDeployment(namespacedName, 1)
+			ns := scheme.Codecs.WithoutConversion()
+			tf := cmdtesting.NewTestFactory().WithNamespace(namespacedName.Namespace)
+			tf.ClientConfigVal = cmdtesting.DefaultClientConfig()
+
+			info, _ := runtime.SerializerInfoForMediaType(ns.SupportedMediaTypes(), runtime.ContentTypeJSON)
+			encoder := ns.EncoderForVersion(info.Serializer, rolloutGroupVersionEncoder)
+			tf.Client = &RolloutRestartRESTClient{
+				RESTClient: &fake.RESTClient{
+					GroupVersion:         rolloutGroupVersionEncoder,
+					NegotiatedSerializer: ns,
+					Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+						responseDeployment := &appsv1.Deployment{}
+						responseDeployment.Name = deployment1.Name
+						responseDeployment.Labels = make(map[string]string)
+						responseDeployment.Labels[tc.labelKey] = tc.labelValue
+						body := io.NopCloser(bytes.NewReader([]byte(runtime.EncodeOrDie(encoder, responseDeployment))))
+						return &http.Response{StatusCode: http.StatusOK, Header: cmdtesting.DefaultHeader(), Body: body}, nil
+					}),
+				},
+			}
+			tf.FakeDynamicClient.WatchReactionChain = nil
+			tf.FakeDynamicClient.AddWatchReactor("*", func(action cgtesting.Action) (handled bool, ret watch.Interface, err error) {
+				fw := watch.NewFake()
+				dep := &appsv1.Deployment{}
+				dep.Name = deployment1.Name
+				dep.Status = appsv1.DeploymentStatus{
+					Replicas:            1,
+					UpdatedReplicas:     1,
+					ReadyReplicas:       1,
+					AvailableReplicas:   1,
+					UnavailableReplicas: 0,
+					Conditions: []appsv1.DeploymentCondition{{
+						Type: appsv1.DeploymentAvailable,
+					}},
+				}
+				dep.Labels = make(map[string]string)
+				dep.Labels[tc.labelKey] = tc.labelValue
+				c, err := runtime.DefaultUnstructuredConverter.ToUnstructured(dep.DeepCopyObject())
+				if err != nil {
+					t.Errorf("unexpected err %s", err)
+				}
+				u := &unstructured.Unstructured{}
+				u.SetUnstructuredContent(c)
+				go fw.Add(u)
+				return true, fw, nil
```

Comment:
Let's record for which resources (and how many times) this was called

Reply:
updated the number of times